Error while checking build/O2Physics/code-check for a0b1ba3f05aaf9c1df1eefd74052683e0e0088ec at 2026-08-17 23:00:
```
## sw/BUILD/O2Physics-code-check-latest/log
--
========== List of issues found ==========
++ echo 'Found 16 errors and 340 warnings.'
Found 16 errors and 340 warnings.
++ [[ 16 -gt 0 ]]
++ cat /sw/BUILD/3095f7b4dd35a1796d16378b6c1fedc17af8ee7f/O2Physics-code-check/errors.txt
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:75:8: error: constructor does not initialize these fields: ccdb [cppcoreguidelines-pro-type-member-init,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:356:9: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:357:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:358:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:359:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:360:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:361:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:362:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:363:11: error: implicit conversion 'bool' -> 'int' [readability-implicit-bool-conversion,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1489:5: error: redundant return statement at the end of a function with a void return type [readability-redundant-control-flow,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1838:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1857:22: error: do not declare C-style arrays, use 'std::array' instead [modernize-avoid-c-arrays,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:2183:7: error: do not use 'else' after 'return' [readability-else-after-return,-warnings-as-errors]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:3092:31: error: use 'contains' to check for membership [readability-container-contains,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx:37:1: error: included header Logger.h is not used directly [misc-include-cleaner,-warnings-as-errors]
PWGLF/Tasks/Resonances/k892hadronphoton.cxx:813:5: error: if with identical then and else branches [bugprone-branch-clone,-warnings-as-errors]
++ [[ 340 -gt 0 ]]
++ cat /sw/BUILD/3095f7b4dd35a1796d16378b6c1fedc17af8ee7f/O2Physics-code-check/warnings.txt
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:459:64: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:918:224: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:922:274: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1069:67: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1071:68: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1073:68: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1077:91: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1079:92: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1081:92: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1165:26: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1171:47: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1176:33: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1255:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1260:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1265:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1270:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1275:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1280:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1285:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1290:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1295:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1300:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1305:19: warning: statement should be inside braces [readability-braces-around-statements]
PWGLF/TableProducer/Strangeness/sigma0builder.cxx:1310:19: warning: statement should be inside braces [readability-braces-around-statements]
[0 more errors; see full log]
```
Full log [here](https://ali-ci.cern.ch/alice-build-logs/AliceO2Group/O2Physics/17480/a0b1ba3f05aaf9c1df1eefd74052683e0e0088ec/build_O2Physics_code-check/pretty.html).

--- a/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
+++ b/PWGLF/TableProducer/Strangeness/sigma0builder.cxx
@@ -1622,7 +1622,7 @@ struct sigma0builder {
     GenInfo.IsPrimary = mcParticle.isPhysicalPrimary();
     GenInfo.IsV0Lambda = mcParticle.pdgCode() == PDG_t::kLambda0;                                // 3122
     GenInfo.IsV0AntiLambda = mcParticle.pdgCode() == PDG_t::kLambda0Bar;                         //-3122
-    GenInfo.IsV0KShort = mcParticle.pdgCode() == PDG_t::kK0Short;                                // 310 // also abs ?
+    GenInfo.IsV0KShort = std::abs(mcParticle.pdgCode()) == PDG_t::kK0Short;                      // 310
     GenInfo.IsPi0 = mcParticle.pdgCode() == PDG_t::kPi0;                                         // 111;
     GenInfo.IsSigma0 = mcParticle.pdgCode() == PDG_t::kSigma0;                                   // PDG_t::kSigma0
     GenInfo.IsAntiSigma0 = mcParticle.pdgCode() == PDG_t::kSigma0Bar;                            //-3212

--- a/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphoton.cxx
@@ -809,6 +809,8 @@ struct k892hadronphoton {
     int trkCode = 10; // 1: TPC-only, 2: TPC+Something, 3: ITS-Only, 4: ITS+TPC + Something, 10: anything else
     auto photonPosTrackCode = kstar.photonPosTrackCode();
     auto photonNegTrackCode = kstar.photonNegTrackCode();
+    auto kshortPosTrackCode = kstar.kshortPosTrackCode();
+    auto kshortNegTrackCode = kstar.kshortNegTrackCode();
 
     if (isGamma) {
       if (photonPosTrackCode == DauTPCOnly && photonNegTrackCode == DauTPCOnly)
@@ -820,13 +822,13 @@ struct k892hadronphoton {
       if (photonPosTrackCode == DauITSTrackerOnly || photonNegTrackCode == DauITSTrackerOnly)
         trkCode = 4;
     } else {
-      if (photonPosTrackCode == DauTPCOnly && photonNegTrackCode == DauTPCOnly)
+      if (kshortPosTrackCode == DauTPCOnly && kshortNegTrackCode == DauTPCOnly)
         trkCode = 1;
-      if ((photonPosTrackCode != DauTPCOnly && photonNegTrackCode == DauTPCOnly) || (photonPosTrackCode == DauTPCOnly && photonNegTrackCode != DauTPCOnly))
+      if ((kshortPosTrackCode != DauTPCOnly && kshortNegTrackCode == DauTPCOnly) || (kshortPosTrackCode == DauTPCOnly && kshortNegTrackCode != DauTPCOnly))
         trkCode = 2;
-      if (photonPosTrackCode == DauITSTPC && photonNegTrackCode == DauITSTPC)
+      if (kshortPosTrackCode == DauITSTPC && kshortNegTrackCode == DauITSTPC)
         trkCode = 3;
-      if (photonPosTrackCode == DauITSTrackerOnly || photonNegTrackCode == DauITSTrackerOnly)
+      if (kshortPosTrackCode == DauITSTrackerOnly || kshortNegTrackCode == DauITSTrackerOnly)
         trkCode = 4;
     }
 

--- a/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
+++ b/PWGLF/Tasks/Resonances/k892hadronphotonBkg.cxx
@@ -34,7 +34,6 @@
 #include <Framework/HistogramRegistry.h>
 #include <Framework/HistogramSpec.h>
 #include <Framework/InitContext.h>
-#include <Framework/Logger.h>
 #include <Framework/OutputObjHeader.h>
 #include <Framework/runDataProcessing.h>
 

